gadget/layout.go

```diff
@@ -299,7 +299,7 @@ func resolveVolumeContent(gadgetRootDir, kernelRootDir string, kernelInfo *kerne
 	for idx := range ps.Content {
 		resolvedSource, err := resolveContentPathOrRef(gadgetRootDir, kernelRootDir, kernelInfo, ps.Content[idx].UnresolvedSource)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("cannot resolve content for structure %v at index %v: %v", ps, idx, err)
```

Comment:
content also has String() defined for it:

```suggestion
			return nil, fmt.Errorf("cannot resolve structure %v content %v: %v", ps, ps.Content[idx], err)
```

The error will then look like:
```
cannot resolve structure #0 content source:$kernel:-invalid-kernel-ref/boot-assets/:  ..
```